Features/source generator autoimplement

luizfbicalho opened the pull request:
Closes #4098

StefanOssendorf commented:
I have only skimmed over some files. But there seem to be a lot of xml-comment c&p errors. Especially on classes which are used by users. Please double check that.

luizfbicalho commented:
> I have only skimmed over some files. But there seem to be a lot of xml-comment c&p errors. Especially on classes which are used by users. Please double check that.

Sorry about that , I had it on my list and checked it as done wrongly, I'll do that later

luizfbicalho commented:
> I have only skimmed over some files. But there seem to be a lot of xml-comment c&p errors. Especially on classes which are used by users. Please double check that.

I changed the comments, do you think that we should change the namespace of the attribute? or some other namespaces?

rockfordlhotka commented on Source/Csla.Generators/cs/AutoImplementProperties/readme.md:
Is this file accurate for the latest AutoImplementProperties behaviors?

luizfbicalho replied:
This I forgot, sending now

rockfordlhotka commented on Source/Csla.Generators/cs/AutoImplementProperties/Csla.Generator.AutoImplementProperties.CSharp/Internals/HashCode.cs:
We can't use the built-in hashcode capabilities of .NET?

Also, just to confirm, this is original code, and not something brought in from another source?

luizfbicalho replied:
This is a gift from @StefanOssendorf

rockfordlhotka replied:
I'm fine with it as long as the IP is clearly available for use in CSLA. Can you confirm that @StefanOssendorf ?

ossendorf-at-hoelscher replied:
I don't think it's my IP I guess?
I took it from here (there's also a comment in my gen that it's taken from there) https://github.com/andrewlock/StronglyTypedId/blob/master/src/StronglyTypedIds/HashCode.cs based/recommended from the author here https://andrewlock.net/creating-a-source-generator-part-9-avoiding-performance-pitfalls-in-incremental-generators/#4-watch-out-for-collection-types

ossendorf-at-hoelscher replied:
In that regard maybe we should consider https://www.nuget.org/packages/Microsoft.Bcl.HashCode/ as a substitute?

rockfordlhotka replied:
I think that is best, please make that change @luizfbicalho.

rockfordlhotka commented on Source/Csla.Generators/cs/AutoImplementProperties/Csla.Generator.AutoImplementProperties.CSharp/Internals/HashCode.cs:
```diff
@@ -0,0 +1,371 @@
+﻿using System.ComponentModel;
```
Missing standard file header/copyright

rockfordlhotka commented on Source/Csla.Generators/cs/AutoImplementProperties/Csla.Generator.AutoImplementProperties.CSharp/AutoImplement/Discovery/DefinitionExtractionContext.cs:
A general question I have with all the support classes that abstract Roslyn: should these be part of this specific generator, or are they useful for _any_ generator and should be their own project/dll that can be reused in the future?

luizfbicalho replied:
I think that this is very specific, I copied from the other generator but it's like a linq query over a database, it's the view model that I need here.
surely they are similar but they should evolve not in the same direction

luizfbicalho commented:
> > > I have only skimmed over some files. But there seem to be a lot of xml-comment c&p errors. Especially on classes which are used by users. Please double check that.
> > 
> > 
> > I changed the comments, do you think that we should change the namespace of the attribute? or some other namespaces?
> 
> I think the attributes should probably be in the `Csla` namespace so they are available by default if someone has added the generator package to their project.

Changed to Csla

luizfbicalho commented:
Updated
